prow/scripts/cluster-integration/kyma-integration-gardener-api-gateway.sh

```diff
@@ -0,0 +1,176 @@
+# - GARDENER_REGION - Gardener compute region
+# - GARDENER_ZONES - Gardener compute zones inside the region
+# - GARDENER_CLUSTER_VERSION - Version of the Kubernetes cluster
+# - GARDENER_KYMA_PROW_KUBECONFIG - Kubeconfig of the Gardener service account
+# - GARDENER_KYMA_PROW_PROJECT_NAME - Name of the gardener project where the cluster will be integrated.
+# - GARDENER_KYMA_PROW_PROVIDER_SECRET_NAME - Name of the secret configured in the gardener project to access the cloud provider
+# - MACHINE_TYPE - (optional) machine type
+#
+#Please look in each provider script for provider specific requirements
+
+
+
+
+# exit on error, and raise error when variable is not set when used
+set -e
+
+ENABLE_TEST_LOG_COLLECTOR=false
+
+export TEST_INFRA_SOURCES_DIR="${KYMA_PROJECT_DIR}/test-infra"
+export KYMA_SOURCES_DIR="${KYMA_PROJECT_DIR}/kyma"
+export TEST_INFRA_CLUSTER_INTEGRATION_SCRIPTS="${TEST_INFRA_SOURCES_DIR}/prow/scripts/cluster-integration/helpers"
+
+# shellcheck source=prow/scripts/lib/log.sh
+source "${TEST_INFRA_SOURCES_DIR}/prow/scripts/lib/log.sh"
+# shellcheck source=prow/scripts/lib/utils.sh
+source "${TEST_INFRA_SOURCES_DIR}/prow/scripts/lib/utils.sh"
+# shellcheck source=prow/scripts/lib/kyma.sh
+source "${TEST_INFRA_SOURCES_DIR}/prow/scripts/lib/kyma.sh"
+
+# All provides require these values, each of them may check for additional variables
+requiredVars=(
+    GARDENER_PROVIDER
+    KYMA_PROJECT_DIR
+    GARDENER_REGION
+    GARDENER_ZONES
+    GARDENER_CLUSTER_VERSION
+    GARDENER_KYMA_PROW_KUBECONFIG
+    GARDENER_KYMA_PROW_PROJECT_NAME
+    GARDENER_KYMA_PROW_PROVIDER_SECRET_NAME
+)
+
+utils::check_required_vars "${requiredVars[@]}"
+
+if [[ $GARDENER_PROVIDER == "azure" ]]; then
+    # shellcheck source=prow/scripts/lib/gardener/azure.sh
+    source "${TEST_INFRA_SOURCES_DIR}/prow/scripts/lib/gardener/azure.sh"
+elif [[ $GARDENER_PROVIDER == "aws" ]]; then
+    # shellcheck source=prow/scripts/lib/gardener/aws.sh
+    source "${TEST_INFRA_SOURCES_DIR}/prow/scripts/lib/gardener/aws.sh"
+elif [[ $GARDENER_PROVIDER == "gcp" ]]; then
+    # shellcheck source=prow/scripts/lib/gardener/gcp.sh
+    source "${TEST_INFRA_SOURCES_DIR}/prow/scripts/lib/gardener/gcp.sh"
+else
+    ## TODO what should I put here? Is this a backend?
+    log::error "GARDENER_PROVIDER ${GARDENER_PROVIDER} is not yet supported"
+    exit 1
+fi
+
+# nice cleanup on exit, be it succesful or on fail
+trap gardener::cleanup EXIT INT
+
+#Used to detect errors for logging purposes
+ERROR_LOGGING_GUARD="true"
+export ERROR_LOGGING_GUARD
+
+RANDOM_NAME_SUFFIX=$(LC_ALL=C tr -dc 'a-z0-9' < /dev/urandom | head -c6)
+readonly COMMON_NAME_PREFIX="grd"
+COMMON_NAME=$(echo "${COMMON_NAME_PREFIX}${RANDOM_NAME_SUFFIX}" | tr "[:upper:]" "[:lower:]")
+export COMMON_NAME
+
+### Cluster name must be less than 10 characters!
+export CLUSTER_NAME="${COMMON_NAME}"
+
+# set KYMA_SOURCE used by gardener::install_kyma
```

Comment:
is it used by `kyma alpha deploy`?

Reply:
no indeed, I will remove that leftover from there